### DISCUSSION TITLE
Version Checking compatible with docker

### DISCUSSION THREAD

**Discussion by Iliescu-Dorin** [containerd/nerdctl#2085] [Q&A] [answered]
### What is the problem you're trying to solve

I've just tried using nerdctl instead the dockercli in remote containers and it fails during checking the version of the docker api:
level=fatal msg="template: :1:9: executing "" at <.Server.APIVersion>: can't evaluate field APIVersion in type *dockercompat.ServerVersion"
After some trail & error, I could fetch the server version with the following command:
nerdctl version --format '{{(index .Server.Components 0).Version}}'

This info was taken from [this](https://github.com/microsoft/vscode-remote-release/issues/6252#issuecomment-1114569495) comment

### Describe the solution you'd like

I wonder, whether you guys could add extra configuration entries like Version format template (so we can configure the version template thingy) and also making the minimum version configurable, I could try using Devcontainers + nerdctl.

### Additional context

After this, I've done some digging in moby's source code compared to the nerdctl sourcecode and I've noticed that the Properties .Server.Version are marked deprecated and are kept for backwards compat.:
https://github.com/moby/moby/blob/2ed904cad7055847796433cc56ef1d1de0da868c/api/types/types.go#L219
The nerdctl guys did not implement it in their implementation: https://github.com/containerd/nerdctl/blob/2b4e3bde28d67d08c85bcf15fef23d973bae590f/pkg/inspecttypes/dockercompat/info.go#L95

**Comment by AkihiroSuda** [comment, marked as answer]
containerd does not implement the Docker REST API, so `.Server.APIVersion` doesn't make a sense for containerd.
